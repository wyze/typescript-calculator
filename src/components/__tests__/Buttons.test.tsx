import { h } from 'preact'
import Buttons from '../Buttons'
import snapshot from './utils/snapshot'

const props = {
  actions: {
    add: jest.fn(),
    clear: jest.fn(),
    divide: jest.fn(),
    equals: jest.fn(),
    input: jest.fn(),
    multiply: jest.fn(),
    percent: jest.fn(),
    posneg: jest.fn(),
    subtract: jest.fn(),
  },
}

describe('<Buttons />', () => {
  it('renders', () => snapshot(<Buttons {...props} />))
})
