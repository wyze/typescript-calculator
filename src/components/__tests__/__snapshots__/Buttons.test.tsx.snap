// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Buttons /> renders 1`] = `
<div>
  <ButtonGroup>
    <OperationButton
      action={[Function]}
      className={
        Object {
          "data-css-llx5cd": "",
        }
      }
      text="C"
    />
    <OperationButton
      action={[Function]}
      className={
        Object {
          "data-css-llx5cd": "",
        }
      }
      text="+/−"
    />
    <OperationButton
      action={[Function]}
      className={
        Object {
          "data-css-llx5cd": "",
        }
      }
      text="%"
    />
    <OperationButton
      action={[Function]}
      text="÷"
    />
  </ButtonGroup>
  <ButtonGroup>
    <InputButton
      action={[Function]}
      value="7"
    />
    <InputButton
      action={[Function]}
      value="8"
    />
    <InputButton
      action={[Function]}
      value="9"
    />
    <OperationButton
      action={[Function]}
      text="×"
    />
  </ButtonGroup>
  <ButtonGroup>
    <InputButton
      action={[Function]}
      value="4"
    />
    <InputButton
      action={[Function]}
      value="5"
    />
    <InputButton
      action={[Function]}
      value="6"
    />
    <OperationButton
      action={[Function]}
      text="−"
    />
  </ButtonGroup>
  <ButtonGroup>
    <InputButton
      action={[Function]}
      value="1"
    />
    <InputButton
      action={[Function]}
      value="2"
    />
    <InputButton
      action={[Function]}
      value="3"
    />
    <OperationButton
      action={[Function]}
      text="+"
    />
  </ButtonGroup>
  <ButtonGroup>
    <InputButton
      action={[Function]}
      className={
        Object {
          "data-css-ch1ln3": "",
        }
      }
      value="0"
    />
    <InputButton
      action={[Function]}
      value="."
    />
    <OperationButton
      action={[Function]}
      text="="
    />
  </ButtonGroup>
</div>
`;
